authorization/src/main/kotlin/com/ritense/authorization/permission/condition/ReflectingPermissionCondition.kt

```diff
@@ -16,12 +16,25 @@
 
 package com.ritense.authorization.permission.condition
 
+import java.lang.reflect.Field
+
 abstract class ReflectingPermissionCondition(type: PermissionConditionType) : PermissionCondition(type) {
     protected fun findEntityFieldValue(entity: Any, field: String): Any? {
         var currentEntity: Any? = entity
         val fields = field.split('.')
         fields.forEachIndexed { index, value ->
-            val declaredField = currentEntity!!.javaClass.getDeclaredField(value)
+            var classToSearch: Class<*>? = entity.javaClass
+            var declaredField: Field? = null
+
+            while (declaredField == null && classToSearch != null) {
+                declaredField = classToSearch.declaredFields.firstOrNull { it.name == value }
```

Comment:
Maybe not part of this story perse since it was already there, but why are we checking declared fields instead of just the public fields? This means we can never change a private field without it effectively being a breaking change.